### PR TITLE
An 5001/update amm sell

### PR DESCRIPTION
**Pull request by tarikceric**
Use decoded instructions to create new `nft_sales_amm_sell_decoded` model
- deprecate legacy model `nft_sales_amm_sell` and create view
- use legacy model for sales before 2022-10-30 in gold `fact_nft_sales` because those txs cannot be decoded
- add relationship test to decoded_instructions_combined

```
full refresh
1 of 1 OK created sql incremental model silver.nft_sales_amm_sell_decoded ...... [SUCCESS 1 in 929.20s]

incremental
1 of 1 OK created sql incremental model silver.nft_sales_amm_sell_decoded ...... [SUCCESS 244 in 21.26s]

 test
01:25:51  Finished running 18 tests, 11 hooks in 0 hours 0 minutes and 37.88 seconds (37.88s).
01:25:51  
01:25:51  Completed successfully
01:25:51  
01:25:51  Done. PASS=18 WARN=0 ERROR=0 SKIP=0 TOTAL=18

nft.fact_nft_sales incremental
1 of 1 OK created sql view model silver.nft_sales_amm_sell_view ................ [SUCCESS 1 in 1.64s]
```

### CODE REVIEW

**Review by desmond-hui — CHANGES_REQUESTED** [review]
Getting `NULL` sales_amount when I run the test on the full output

```
recent_date_filter: &recent_date_filter
      config:
        where: _inserted_timestamp >= '2020-01-01'
```

```
16:11:58  Completed with 1 error and 0 warnings:
16:11:58  
16:11:58  Failure in test not_null_silver__nft_sales_amm_sell_decoded_SALES_AMOUNT (models/silver/nfts/silver__nft_sales_amm_sell_decoded.yml)
16:11:58    Got 27 results, configured to fail if != 0
16:11:58  
16:11:58    compiled Code at target/compiled/solana_models/models/silver/nfts/silver__nft_sales_amm_sell_decoded.yml/not_null_silver__nft_sales_amm_sell_decoded_SALES_AMOUNT.sql
16:11:58  
16:11:58    See test failures:
  --------------------------------------------------------------------------------
  select * from SOLANA_DEV.not_null_silver.nft_sales_amm_sell_decoded_SALES_AMOUNT
  --------------------------------------------------------------------------------
16:11:58  
16:11:58  Done. PASS=17 WARN=0 ERROR=1 SKIP=0 TOTAL=18
```

**Review by desmond-hui — COMMENTED** [review]
```
depositSell
extDepositSell
mip1DepositSell
ocpDepositSell
solDepositBuy
```

What do these events do if they are not sales?

**Comment by tarikceric** [comment]
> ```
> depositSell
> extDepositSell
> mip1DepositSell
> ocpDepositSell
> solDepositBuy
> ```
> 
> What do these events do if they are not sales?

They're related to creation of liquidity pools and depositing NFTs into them (or depositing SOL into a buy-side liquidity pool for solDepositBuy). Don't completely understand how it works but those events don't seem to deal with the direct NFT sales.

**Comment by tarikceric** [comment]
> Getting `NULL` sales_amount when I run the test on the full output
> 
> ```
> recent_date_filter: &recent_date_filter
>       config:
>         where: _inserted_timestamp >= '2020-01-01'
> ```
> 
> ```
> 16:11:58  Completed with 1 error and 0 warnings:
> 16:11:58  
> 16:11:58  Failure in test not_null_silver__nft_sales_amm_sell_decoded_SALES_AMOUNT (models/silver/nfts/silver__nft_sales_amm_sell_decoded.yml)
> 16:11:58    Got 27 results, configured to fail if != 0
> 16:11:58  
> 16:11:58    compiled Code at target/compiled/solana_models/models/silver/nfts/silver__nft_sales_amm_sell_decoded.yml/not_null_silver__nft_sales_amm_sell_decoded_SALES_AMOUNT.sql
> 16:11:58  
> 16:11:58    See test failures:
>   --------------------------------------------------------------------------------
>   select * from SOLANA_DEV.not_null_silver.nft_sales_amm_sell_decoded_SALES_AMOUNT
>   --------------------------------------------------------------------------------
> 16:11:58  
> 16:11:58  Done. PASS=17 WARN=0 ERROR=1 SKIP=0 TOTAL=18
> ```

Good catch, didn't think about the filter. I added additional event_type in the ME logic, and the test on the complete dataset is passing:

```
19:55:21  Finished running 18 tests, 11 hooks in 0 hours 0 minutes and 41.24 seconds (41.24s).
19:55:21  
19:55:21  Completed successfully
19:55:21  
19:55:21  Done. PASS=18 WARN=0 ERROR=0 SKIP=0 TOTAL=18
```

File: models/silver/nfts/silver__nft_sales_amm_sell_decoded.sql [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+                decoded_instruction :accounts
+            )
+        END AS purchaser,
+        CASE
+            WHEN event_type IN (
+                'solFulfillSell',
+                'solMip1FulfillSell',
+                'solOcpFulfillSell'
+            ) THEN silver.udf_get_account_pubkey_by_name(
+                'pool',
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_amm_sell_decoded.sql:109]: The seller is the NFT pool for coral cube?

**Review by desmond-hui — APPROVED** [review]

File: models/silver/nfts/silver__nft_sales_amm_sell_decoded.sql [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+            WHEN event_type IN (
+                'solFulfillSell',
+                'solMip1FulfillSell',
+                'solOcpFulfillSell'
+            ) THEN silver.udf_get_account_pubkey_by_name(
+                'payer',
+                decoded_instruction :accounts
+            )
+            ELSE silver.udf_get_account_pubkey_by_name(
+                'pool',
```
[the remark this answers is not in the document]
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_amm_sell_decoded.sql:99]: in these instances the 'pool' address is what recieves the NFT and has ownership, and also what sends payment so we ID that as technically the purchaser.

File: models/silver/nfts/silver__nft_sales_amm_sell_decoded.sql (hunk 1 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_amm_sell_decoded.sql:109]: vice versa for purchaser comment

**Pull request merged**